batch.go

```diff
@@ -243,14 +233,30 @@ func newLabelPool(faker *fake.Faker, cardinalities map[string]int) LabelPool {
 		"format": []string{"apache_common", "apache_combined", "apache_error", "rfc3164", "rfc5424", "json", "logfmt"}, // needs to match the available flog formats
 		"os":     []string{"darwin", "linux", "windows"},
 	}
+
+	num := 10
+	var nss []string
 	if n, ok := cardinalities["namespace"]; ok {
-		lb["namespace"] = generateValues(faker.BS, n)
+		num = n
 	}
+	nss = generateValues(faker.BS, num)
+	lb["namespace"] = nss
+	num = 5
 	if n, ok := cardinalities["app"]; ok {
-		lb["app"] = generateValues(faker.AppName, n)
+		num = n
 	}
+	svs := generateValues(fakeAppName(faker), num)
+	lb["app"] = svs
+	var jobs []string
+	for _, ns := range nss {
```

Comment:
just guessing, but this loop has potential for duplicate labels

Reply:
These loops create `job` labels from generated namespaces and services and join them to the namespace/service pattern. But assigning `app` in 249 could duplicate labels, so I deleted it.